app/uk/gov/hmrc/tai/service/PaymentsService.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tai.service
+
+import javax.inject.Inject
+import uk.gov.hmrc.tai.model.domain.{Employment, Payment}
+import uk.gov.hmrc.tai.viewModels.PaymentDetailsViewModel
+
+class PaymentsService @Inject()() {
+  def filterDuplicates(employment: Employment): Seq[PaymentDetailsViewModel] = {
+    val payments = employment.latestAnnualAccount.map(_.payments).getOrElse(Seq.empty[Payment])
```

Comment:
Is payments on `latestAnnualAccount` really an `Option[List[Payment]]`. If it isn't too much work consider refactoring it to be `List[Payment]` instead as all the semantics of an `Option` can be realized with a `List` so `Option[List[_]]` doesn't really make sense

Reply:
It grabs the latest annual account (the list.max) but if it's empty it's a None, so really it's just an isEmpty wrapper that turns it into a None/AnnualAccount. 

`lazy val latestAnnualAccount: Option[AnnualAccount] = if (annualAccounts.isEmpty) None else Some (annualAccounts.max)`